### PR TITLE
Automatically install dependencies when package is installed

### PR DESCRIPTION
**Pull request by sohang3112**
All dependencies (NumPy, SciPy, Matplotlib, OpenCV) will be installed automatically when `imutils` is installed.

This is done using `setuptools`, since `distutils` doesn't support automatic installation of dependencies.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -12,4 +18,10 @@
     keywords=['computer vision', 'image processing', 'opencv', 'matplotlib'],
     classifiers=[],
     scripts=['bin/range-detector'],
+    install_requires=[
+        'numpy', 
+        'scipy',
```
Comment (farqis) [inline setup.py:23]: I have raised a PR to remove scipy dependency. If that gets approved then you might have one less dependency to install.

https://github.com/PyImageSearch/imutils/pull/277#issue-1437785798

Comment (sohang3112) [reply setup.py:23]: @farqis When your PR is merged, please update that here and resolve this conversation.

Comment (sohang3112) [reply setup.py:23]: @farqis Was your PR merged (for removing scipy dependency)?

**Comment by sohang3112** [comment]
@jrosebr1 Please review my pull request - it will make installing `imutils` easier. Let me know if something can be improved in the PR. Thanks in advance 🙂.

**Comment by sohang3112** [comment]
@jrosebr1 Did you have a chance to review my PR yet?

**Comment by jrosebr1** [comment]
@sohang3112 I no longer maintain this library. Please contact PyImageSearch support. Thank you.

**Comment by sohang3112** [comment]
>  I no longer maintain this library.

Who is the current maintainer?

> Please contact PyImageSearch support. 

I filled this contact form of PyImageSearch: https://pyimagesearch.com/contact/ . Should I contact them in some other way also?

**Comment by jrosebr1** [comment]
@sohang3112 Yes, use the contact form. Thanks for submitting the PR, I hope it's merged 🙏

**Comment by sohang3112** [comment]
@jrosebr1 I filled the contact form, but I got this email in response:

> Because of the amount of correspondence we receive, we only have time to provide technical advice to our current customers right now. For everyone else, we have hundreds of free tutorials on our site. We hope you understand.

I'm not going to buy their course just to get my PR merged 🙂

**Comment by sohang3112** [comment]
Closing this PR, since @jrosebr1 no longer maintains the repo and PyImageSearch did not respond at all.

File: setup.py (hunk 1 above)
Comment (farqis) [reply setup.py:23]: > @farqis Was your PR merged (for removing scipy dependency)?

@sohang3112 
3 and half years later...
Still not merged. 😅

**Pull request closed without merging**